api/kyverno/v1alpha1/cleanup_policy_types.go

```diff
@@ -109,3 +121,185 @@ type CleanupPolicySpec struct {
 type CleanupPolicyStatus struct {
 	Conditions []metav1.Condition `json:"conditions,omitempty" patchStrategy:"merge" patchMergeKey:"type" protobuf:"bytes,1,rep,name=conditions"`
 }
+
+// Validate implements programmatic validation
+func (p *CleanupPolicySpec) Validate(clusterResources sets.String, namespaced bool) (errs field.ErrorList) {
+	errs = append(errs, ValidateSchedule(field.NewPath("spec").Child("schedule"), p.Schedule)...)
+	errs = append(errs, p.MatchResources.Validate(field.NewPath("spec").Child("match"), namespaced, clusterResources)...)
+	errs = append(errs, p.ExcludeResources.Validate(field.NewPath("spec").Child("exclude"), namespaced, clusterResources)...)
+	errs = append(errs, p.ValidateMatchExcludeConflict(field.NewPath("spec"))...)
+	return errs
+}
+
+// ValidateSchedule validates whether the schedule specified is in proper cron format or not.
+func ValidateSchedule(path *field.Path, schedule string) (errs field.ErrorList) {
+	if _, err := cron.ParseStandard(schedule); err != nil {
+		errs = append(errs, field.Invalid(path, schedule, "schedule spec in the cleanupPolicy is not in proper cron format"))
+	}
+	return errs
+}
+
+// ValidateMatchExcludeConflict checks if the resultant of match and exclude block is not an empty set
+func (spec *CleanupPolicySpec) ValidateMatchExcludeConflict(path *field.Path) (errs field.ErrorList) {
```

Comment:
Can't we reuse those from `v1` ?

Reply:
That function is made for type `Rule`, not for `CleanupPolicySpec`. So we cannot use that function, right?

Reply:
Ok, we should find a way to factor them out but we can do it later in a follow up PR.